### PR TITLE
Throw exception if Public User data is not found

### PR DESCRIPTION
**Pull request by jtpalmer**
## Description

See title.  Any suggestions for a better error message are welcome.

## Motivation and Context

If the public user data is missing, the portal will not be displayed and a blank screen is presented instead.  This happens if the database is not migrated from 6.6. to 7.0.  This PR changes the behaviour to display:

```
{"success":false,"count":0,"total":0,"totalCount":0,"results":[],"data":[],"message":"Public User not found","code":0}
```

Which at least includes an error message.

## Tests performed

- Installed 6.6
- Upgraded to 7.0
- (Did not run `xdmod-upgrade`)
- Loaded portal
- Observed error message

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jtpalmer** [comment]
Moved exception to the `getUserByUserName` method after discussion with @plessbd .

**Comment by jpwhite4** [comment]
Please can you add a section to the FAQ (either here, or a a different pull request) that explains what to do if you see this message.

**Comment by plessbd** [comment]
Maybe have the error point to the FAQ?

File: open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php [hunk 1]
```diff
@@ -424,10 +424,12 @@ public function testGetUserByUserNameEmptyString()
         $this->assertNull($user);
     }
 
+    /**
+     * @expectedException Exception
+     */
     public function testGetUserByUserNameNull()
```
Comment (plessbd) [inline open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php]: should we add a test for `$user = XDUser::getUserByUserName('Ben is not really a user here...');`

File: open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php [hunk 2]
```diff
@@ -424,10 +424,12 @@ public function testGetUserByUserNameEmptyString()
         $this->assertNull($user);
     }
 
+    /**
+     * @expectedException Exception
```
Comment (plessbd) [inline open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php]: the previous two tests need to have this as well I believe, as they will also now through an exception.

Comment (jtpalmer) [reply open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php]: I've updated those tests now

**Pull request merged**